src/creature.cpp

```diff
@@ -148,7 +152,9 @@ void Creature::checkPath()
 {
 	if (attackedCreature) {
 		const Position& targetPos = attackedCreature->getPosition();
-		if (followPosition != targetPos) {
+		if (followPosition != targetPos || forceUpdateFollowPath) {
+			forceUpdateFollowPath = false;
+
```

Comment:
This will always set the value to `false`:

| followPosition != targetPos | forceUpdateFollowPath | output |
|-|-|-|
| true | true | false |
| true | false | false |
| false | true | false |
| false | false | false |

Reply:
? it should do:

if (followPosition != targetPos) : execute
if (forceUpdateFollowPath) : execute

We always set forceUpdateFollowPath = false because we shouldn't need to update again no matter what.
Other areas in the sources can set forceUpdateFollowPath to true, that's the only reason its there.

Reply:
@gigastar but the current code is setting to false in every possible case (check the truth table). Might as well remove the if.

Reply:
@ranisalt  I am not understanding you man.

The code works as intended. It executes if the player moves or after the onThink event sets forceUpdatePath = true.

The idea is to make sure its not constantly updating the path. Trying to save as much CPU, memory, ect. as possible.

As long as the player doesnt move, the path should only update after the onThink events sets forceUpdatePath.
If the player moves it will read the position is different and execute.

If targetPos changed OR forceUpdateFollowPath is true then
     updateThePath

if targetPos is the same > check if forceupdateFollowPath is true
     -forceUpdateFollowPath is true : execute
     -forceUpdateFollowPath is false : dont execute

We don't want it to update constantly. If we remove the forceUpdateFollowPath check then when players don't move and a monster already has a path, but that path gets blocked. They won't be able to find their way.

There is a misunderstanding here.

Reply:
My bad. I opened this page several times and it was displaying this:

```
		if (followPosition != targetPos || forceUpdateFollowPath) {
			forceUpdateFollowPath = false;
		}
```

Now I see that it has more within the block, idk what was happening.

Reply:
It's probably mad at me for all the commits at this point.